v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/metadata/CassandraSourceMetadata.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.metadata;
+
+import autovalue.shaded.com.google.common.collect.ImmutableList;
+import com.datastax.oss.driver.api.core.cql.ResultSet;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.*;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.cassandra.SourceColumn;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.cassandra.SourceSchema;
+import com.google.cloud.teleport.v2.spanner.migrations.schema.cassandra.SourceTable;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class CassandraSourceMetadata {
+
+  private static final String SCHEMA_NAME = "cassandra"; // Constant for schema name
+
+  /**
+   * Generates a SourceSchema from a ResultSet.
+   *
+   * @param resultSet The ResultSet containing schema information.
+   * @return A SourceSchema instance.
+   */
+  private static SourceSchema generateSourceSchema(ResultSet resultSet) {
+    Map<String, Map<String, SourceColumn>> schema = new HashMap<>();
+
+    resultSet.forEach(
+        row -> {
+          String tableName = row.getString("table_name");
+          String columnName = row.getString("column_name");
+          String dataType = row.getString("type");
+          String kind = row.getString("kind");
+
+          boolean isPrimaryKey = isPrimaryKey(kind);
+          SourceColumn sourceColumn = SourceColumn.create(columnName, kind, dataType, isPrimaryKey);
+
+          schema.computeIfAbsent(tableName, k -> new HashMap<>()).put(columnName, sourceColumn);
+        });
+
+    Map<String, SourceTable> tables =
+        schema.entrySet().stream()
+            .collect(
+                Collectors.toMap(
+                    Map.Entry::getKey,
+                    entry ->
+                        SourceTable.create(
+                            entry.getKey(), ImmutableList.copyOf(entry.getValue().values()))));
+
+    return SourceSchema.create(Map.copyOf(tables));
+  }
+
+  /**
+   * Converts a ResultSet to a Schema object, updating the provided schema.
+   *
+   * @param schema The schema to update.
+   * @param resultSet The ResultSet containing schema information.
+   */
+  public static void generateSourceSchema(Schema schema, ResultSet resultSet) {
```

Comment:
nit: let's just call it for what it does. updateSourceSchema or generateAndSetSourceSchema

Generate seems to suggest something is being returns some value.

Reply:
Address the same @bharadwaj-aditya